File: python/credit_check.py
def credit_check(str):
    #create int array from string
    num_list = [int(num) for num in str]
    sum = 0

    #loop through and double every other
    for i in range(0, len(num_list),2):
        num_list[i] = num_list[i] * 2

    #loop through and sum digits greater than 9
    for i in range(0, len(num_list)):
        if num_list[i] > 9:
            remain = num_list[i] % 10
            num_list[i] = ((num_list[i] - remain)/10) + remain
    
    #loop through and sum 
    for i in range(0, len(num_list)):
        sum += num_list[i]

    #check valid
    if (sum % 10 == 0):
        return "The number is valid!"
    else:
        return "The number is invalid!"
    


# Your Luhn Algorithm Here
# Expected Output:
# If it is valid, print "The number is valid!"
# If it is invalid, print "The number is invalid!"

File: js/creditCheck.js
const creditCheck = (num) => {
    // make array from num input
    arr = Array.from(num.toString()).map(Number)
    let sum = 0
    // loop through array and double every other
    for (i = 0; i < arr.length; i += 2){
        arr[i] = arr[i] * 2
    }
    // loop through array and sum the digits of numbers greater than 9
    for (i = 0; i < arr.length; i++){
        if (arr[i] > 9){
            arr[i] = (Math.floor(arr[i] / 10)) + (arr[i] % 10)
        }
    }
    // loop through array and sum array
    for (i = 0; i < arr.length; i++){
        sum += arr[i]
    }
    // % 10 check for validity
    if (sum % 10 === 0){
        return 'Valid'
    } else {
        return "Invalid"
    }
    
}


console.log(creditCheck(5541808923795240))